crates/floresta-chain/src/pruned_utreexo/consensus.rs

```diff
@@ -238,13 +257,58 @@ impl Consensus {
             ),
         }
     }
-    #[allow(unused)]
+    pub fn validate_block_time(
+        block_timestamp: u32,
+        mtp: u32,
+        time: impl NodeTime,
+    ) -> Result<(), BlockValidationErrors> {
+        if mtp > block_timestamp && block_timestamp > (time.get_time().sub(2 * HOUR)) {
+            return Err(BlockValidationErrors::InvalidBlockTimestamp);
```

Comment:
We should have one error for each. Block too new will eventually become false (after some time has passed). A block that's too old can't become valid ever